### PR TITLE
Add wgpu triangle example

### PR DESCRIPTION
**Pull request by lord**
Combines together the [wgpu triangle example](https://github.com/gfx-rs/wgpu/blob/v0.14.2/wgpu/examples/hello-triangle/main.rs) and the shello example to demonstrate wgpu working with glazier.

Tested on macos and windows.

### CODE REVIEW

File: examples/wgpu-triangle/main.rs [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+    fn prepare_paint(&mut self) {}
+
+    fn paint(&mut self, _: &Region) {
+        let inner = self.inner.as_mut().unwrap();
+        inner.draw();
+        inner.schedule_render();
+    }
+
+    fn idle(&mut self, _: IdleToken) {
```
Comment (jneem) [inline examples/wgpu-triangle/main.rs:193]: Maybe it's worth omitting the methods with default implementations, given that they aren't the focus of the example? Up to you...

**Review by jneem — APPROVED** [review]
I can confirm that it works on linux too.

File: examples/wgpu-triangle/main.rs (hunk 1 above)
Comment (lord) [reply examples/wgpu-triangle/main.rs:190]: good idea! seems like `idle` was actually necessary, but removed the other ones

**Comment by lord** [comment]
Looks like I don't have permissions to merge — would it be possible to merge all these PRs some time this weekend? Have some more changes I'd like to write up, but it's getting hard to keep a branch up to date with all the changes I've already made.

**Comment by raphlinus** [comment]
I'll merge the approved PRs now, and I've also added you.

**Pull request merged**